api_gen/src/main.rs

```diff
@@ -170,6 +189,44 @@ fn generate_bindings(api_dir: &Path) {
         .expect("Couldn't write bindings!");
 }
 
+fn generate_api_loader_bindings(api_dir: &Path) {
+    let wrapper = api_dir.join("include/ffx_api/ffx_api_loader.h");
+
+    let bindings = bindgen(api_dir)
+        .header(wrapper.to_string_lossy())
+        .allowlist_type("PfnFfx\\w+")
+        .allowlist_function("ffx\\w+")
+        .allowlist_type("[fF]fx\\w+")
+        .allowlist_var("FFX\\w+")
+        .generate()
+        .expect("Unable to generate bindings");
+
+    let out_path = Path::new("sys/src");
+    bindings
+        .write_to_file(out_path.join("api_loader_bindings.rs"))
+        .expect("Couldn't write bindings!");
+}
+
+fn generate_upscale_bindings(api_dir: &Path) {
+    let wrapper = api_dir.join("include/ffx_api/ffx_upscale.h");
+
+    let bindings = bindgen(api_dir)
+        .header(wrapper.to_string_lossy())
+        .allowlist_type("PfnFfx\\w+")
+        .allowlist_function("ffx\\w+")
+        .allowlist_type("[fF]fx\\w+")
+        .allowlist_var("FFX\\w+")
+        .bitfield_enum("FfxApiCreateContextUpscaleFlags")
+        .bitfield_enum("FfxApiDispatchFsrUpscaleFlags")
+        .bitfield_enum("FfxApiDispatchUpscaleAutoreactiveFlags")
```

Comment:
This is now leaking/duplicating all "base API" functions and typedefs into the upscaling bindings, making them unusable interchangeably from `mod loader` :/